src/client-side-encryption/auto_encrypter.ts

```diff
@@ -101,6 +105,8 @@ export interface AutoEncryptionOptions {
   proxyOptions?: ProxyOptions;
   /** The TLS options to use connecting to the KMS provider */
   tlsOptions?: CSFLEKMSTlsOptions;
+  /** Options for KMS socket requests. */
+  socketOptions?: ClientEncryptionSocketOptions;
```

Comment:
Is this something we want to be seperately configurable from the client options? I can understand an argument for "yes" because this controls KMS connections vs DB connections. But since this is (at this moment) just for autoFam controls, does it make sense to keep the knob count low and only let this be configured via the client settings?

Reply:
I've refactored this now to always pull the options directly from the client and remove the `socketOptions` property here.